File: src/args.ts
export function awsRegionArgs(region: string): readonly string[] {
  return ["--region", region, "--output", "json"]
}

export function kubectlContextArgs(cluster: string): readonly string[] {
  return ["--context", cluster]
}

export function namespaceArgs(namespace: string | undefined): readonly string[] {
  return namespace === undefined ? [] : ["--namespace", namespace]
}

export function keyValueSetArgs(values: Readonly<Record<string, string | number | boolean>>): readonly string[] {
  return Object.entries(values).flatMap(([key, value]) => ["--set", `${key}=${String(value)}`])
}

File: safe_eks_mcp/args.py
from __future__ import annotations


def aws_region_args(region: str) -> tuple[str, ...]:
    return ("--region", region, "--output", "json")


def kubectl_context_args(cluster: str) -> tuple[str, ...]:
    return ("--context", cluster)


def namespace_args(namespace: str | None) -> tuple[str, ...]:
    return () if namespace is None else ("--namespace", namespace)


def key_value_set_args(values: dict[str, str | int | bool]) -> tuple[str, ...]:
    args: list[str] = []
    for key, value in values.items():
        args.extend(("--set", f"{key}={str(value).lower() if isinstance(value, bool) else value}"))
    return tuple(str(arg) for arg in args)
